### PR TITLE
INT-16 Project Overlay Component

### PR DESCRIPTION
**Pull request by elsiezhu**
## Feature

[TASK: INT-16](https://linear.app/uoftblueprint/issue/INT-16/project-overlay-component)  

Description of feature or bug: Implement the project overlay/modal component according to the Figma design

### Screenshots/Recordings

<details>
  <summary> Click to expand!</summary>

  Screenshot 2024-03-06 at 2 34 50 PM
</details>

### Important Notes

- Implemented project overlay using Mantine's modal component
- Included a sample on the `/projects` page with dummy data from Figma of how the component should be used using Mantine's `useDisclosure` hook

### CODE REVIEW

File: src/project-overlay/ProjectOverlay.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+            <Modal.Header className="header">
+              <Flex direction={"column"}>
+                <Modal.Title className="title">{props.name}</Modal.Title>
+                <Text className="years">{props.years.join(", ")}</Text>
+              </Flex>
+              <Flex gap="xs">
+                <ActionIcon
+                  component="a"
+                  size={45}
+                  href={""}
```
Comment (shaaaaame) [inline src/project-overlay/ProjectOverlay.tsx]: webLink and gitHubLink missing from href I think?

File: src/project-overlay/ProjectOverlay.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+      <Modal
+        opened={props.isOpened}
+        onClose={props.closeModal}
+        size={"xl"}
+        centered
+        withCloseButton={false}
+        radius={15}
+      >
+        <Grid gutter={"sm"}>
+          <Grid.Col span={4}>
```
Comment (shaaaaame) [inline src/project-overlay/ProjectOverlay.tsx]: I think `span={4}` is causing some issue with responsiveness (particularly <420px). Maybe change to `span={12}` if less than some number of pixels?

<details>
  <summary>Image</summary>
  
  image
</details>

File: src/project-overlay/ProjectOverlay.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+          </Grid.Col>
+        </Grid>
+      </Modal>
+    </>
+  );
+}
+
+function TeamsGrid(team: TeamMember[]) {
+  const teamMembers = team.map((member: TeamMember) => (
+    <Grid.Col span={{ base: 10, md: 6, lg: 6 }} key={member.name}>
```
Comment (shaaaaame) [inline src/project-overlay/ProjectOverlay.tsx]: Breakpoints are causing TeamMember names to get cut off around (450px to 600px) and (400px to 320px). Maybe change to `span={{base: 15, sm: 10, md: 6, lg: 6}}` ?


<details>
  <summary>Image</summary>
  
image</details>

**Review by shaaaaame — CHANGES_REQUESTED** [review]
HI ELSIE!!

This looks so so cool! Good job (😼)! I've left a few comments mainly to do with responsiveness. Do let me know if you have any questions :)

File: src/project-overlay/ProjectOverlay.tsx (hunk 1 above)
Comment (elsiezhu) [reply src/project-overlay/ProjectOverlay.tsx]: This is just an example usage! I will add real links when I implement the page :)

Comment (shaaaaame) [reply src/project-overlay/ProjectOverlay.tsx]: oop sorry I wasn't clear, I meant that it's href={""} right now but should it be href={props.webLink}?

Comment (elsiezhu) [reply src/project-overlay/ProjectOverlay.tsx]: Ah, I see! I will add that, thanks for clarifying!!

**Review by shaaaaame — APPROVED** [review]
Perfect Elsie! Great job 🐱

**Review by sarahsonder — APPROVED** [review]
The component is exactly like the Figma design 👀👀👀
It looks good, Elsie!

**Pull request merged**